docs/how_to.rst

```diff
@@ -0,0 +1,102 @@
+======
+How-to
+======
+
+Use twister plugin with regular python tests
+============================================
+
+It is possible to use twister plugin with regular pytest tests to build a source code and run it on a device.
+
+First we have to create a folder (``hello_world``) which contains our tests.
+In this folder we have ``hello_world_test.py`` file which contains code of our tests,
+and ``testspec.yaml`` with a configuration to build scenarios. There is also ``src`` directory
+with our C code which we want to build.
+
+The folder structure can look like that:
+
+.. code-block:: shell
+
+    └── tests
+        └── hello_world
+            ├── CMakeLists.txt
+            ├── hello_world_test.py
+            ├── prj.conf
+            ├── src
+            │   └── main.c
+            └── testspec.yaml
+
+
+Then we have tests implementation like so:
+
+.. code-block:: python
+   :name: hello_world_test.py
+
+    # -- FILE: hello_world_test.py
+    import re
+    import pytest
+
+    # it will generate variants for all available scenarios from yaml file
+    def test_hello_world(specification, builder, dut, log_parser):
+        log_parser.parse(timeout=5)
+        assert log_parser.state == 'PASSED'
+
+    # it will generate variants for scenario1 and scenario2 only
+    @pytest.mark.build_specification('scenario1', 'scenario2')
+    def test_hello_world_2(specification, builder, dut):
+        pattern = re.compile('.*Hello World.*')
+        assert any(pattern.match(line) for line in dut.iter_stdout)
+
+
+In the example above we use ``specification`` fixture to inform ``pytest`` that these tests
+have ``testspec.yaml`` with configuration for building source code.
+Another fixture ``build`` is used to build the source code for our tests.
+Fixture ``dut`` is used to run built code on a device. And last fixture ``log_parser`` is used
+to parse output from a device and search for expected messages (deffined as regex in ``testspec.yaml``)
+
+
+And example test specification:
```

Comment:
I propose move the presentation of `testspec.yaml` earlier - just after showing the folder structure (so at this moment, lines 27-28). I think, that it can be more cleaner to potential users to understand what is used in `@pytest.mark.build_specification`, when they have possibility at first read what `testspec.yaml` file contains.

Reply:
Chaged.